tests/ha/test_server_pod_failure.py

```diff
@@ -505,3 +506,377 @@ def test_degraded_writes_safe_server_pod_shutdown(self):
+                                  args=(event,), kwargs=args)
+        thread.daemon = True  # Daemonize thread
+        thread.start()
+        LOGGER.info("Step 1: Successfully started READs & WRITES in background. Sleeping for %s "
+                    "sec for s3bench setup check.", HA_CFG["common_params"]["30sec_delay"])
+        time.sleep(HA_CFG["common_params"]["30sec_delay"])
+
+        LOGGER.info("Step 2: Shutdown the server pod by deleting deployment (unsafe)")
+        LOGGER.info("Get server pod name to be deleted")
+        pod_list = self.node_master_list[0].get_all_pods(pod_prefix=const.SERVER_POD_NAME_PREFIX)
+        pod_name = random.sample(pod_list, 1)[0]
+        hostname = self.node_master_list[0].get_pod_hostname(pod_name=pod_name)
+
+        LOGGER.info("Deleting server pod %s", pod_name)
+        event.set()
+        resp = self.node_master_list[0].delete_deployment(pod_name=pod_name)
+        LOGGER.debug("Response: %s", resp)
+        assert_utils.assert_false(resp[0],
+                                  f"Failed to delete server pod {pod_name} by deleting deployment")
+        self.deployment_backup = resp[1]
+        self.deployment_name = resp[2]
+        self.restore_pod = self.deploy = True
+        self.restore_method = const.RESTORE_DEPLOYMENT_K8S
+        LOGGER.info(
+            "Step 2: Successfully shutdown/deleted server pod %s by deleting deployment", pod_name)
+
+        LOGGER.info("Step 3: Check cluster status")
+        resp = self.ha_obj.check_cluster_status(self.node_master_list[0])
+        assert_utils.assert_false(resp[0], resp)
+        LOGGER.info(
+            "Step 3: Cluster has some failures due to server pod %s has gone down.", pod_name)
+
+        LOGGER.info("Step 4: Check services status that were running on server pod %s", pod_name)
+        resp = self.hlth_master_list[0].get_pod_svc_status(pod_list=[pod_name], fail=True,
+                                                           hostname=hostname)
+        LOGGER.debug("Response: %s", resp)
+        assert_utils.assert_true(resp[0], resp)
+        LOGGER.info("Step 4: Services of server pod %s are in offline state", pod_name)
+
+        pod_list.remove(pod_name)
+        LOGGER.info("Step 5: Check services status on remaining pods %s", pod_list)
+        resp = self.hlth_master_list[0].get_pod_svc_status(pod_list=pod_list, fail=False)
+        LOGGER.debug("Response: %s", resp)
+        assert_utils.assert_true(resp[0], resp)
+        LOGGER.info("Step 5: Services status on remaining pods are in online state")
+
+        event.clear()
+        thread.join()
+        responses = {}
+        while len(responses) != 2:
+            responses = output.get(timeout=HA_CFG["common_params"]["60sec_delay"])
+        LOGGER.info("Step 6: Verify status for In-flight READs & WRITEs while server pod %s is "
+                    "going down should be failed/error.", pod_name)
```

Comment:
Why failed/error?

Reply:
> Why failed/error?

we are expecting failures in READ/WRITE when server pod was going down. Hence Failed or error

Reply:
Yes for server pods going down, in-flight IOs are expected to fail and need to be retried from s3 client.

Reply:
I was saying why both the words are added as failed/error both depict same thing. Let it be. :)